package-spotlight-class-metadata.md

```diff
@@ -0,0 +1,210 @@
+# Package Spotlight: `zenstruck/class-metadata`
+
+Consider an *auditing* feature in your app. You want to track activity including
+changes to all your entities. Your `activity` table includes a reference to the
+entity (the entity class name) and the entity's ID. This enables you to later
+retrieve and display:
+
+1. All activity
+2. Activity for a specific entity class
+3. Activity for a specific entity instance
+
+Storing the class name in the database has a few downsides:
+
+1. Renaming the class requires a database migration to update potentially 1000's of rows.
+2. This name isn't human-readable. Being human myself, it would be nice if this activity
+   was displayed in a way that was easy for me to read.
+
+Ok, what about showing the short name of the class? `App\Entity\ProductCategory` could be
+`ProductCategory` or even converted to kebab-case: `product-category`. The problem here is you
+can't go the other way: from `product-category` back to `App\Entity\ProductCategory` to be able
+to query the database for *all* `product-category` activity.
+
+## Enter `zenstruck/class-metadata`
+
+First, install it with:
+
+```terminal
+composer require zenstruck/class-metadata
+```
+
+***NOTE
+You'll be prompted to enable the Composer plugin - be sure to do this.
+***
+
+This package enables you to add an `#[Alias]` attribute to any class in your app:
+
+```php
+use Zenstruck\Alias;
+
+#[Alias('product-category')]
+class ProductCategory
+{
+    // ...
+}
+```
+
+***IMPORTANT
+Whenever you add or change an attribute, be sure to run:
+
+```terminal
+composer dump-autoload
+```
+
+Behind the scenes, the Composer plugin hooks into the autoload dump process. It scans
+your codebase for any `#[Alias]` attributes and generates a mapping file that enables
+superfast lookups. The mapping is pure PHP arrays, so no reflection or scanning is
+needed at runtime. But, this does mean you need to run `composer dump-autoload`
+whenever you *add or change an alias*.
+***
+
+## Aliases at Runtime
+
+You can get the alias for a class or object like so:
+
+```php
+use Zenstruck\Alias;
+
+Alias::for(ProductCategory::class); // "product-category"
+Alias::for(new ProductCategory()); // "product-category"
+```
+
+In our auditing example, you'd use this when saving activity to the database:
+
+```php
+function logActivity(object $entity, int $entityId, string $action): void
+{
+    if (!$alias = Alias::for($entity)) {
+        return; // no alias 
+    }
+
+    $this->saveToDatabase($alias, $entityId, $action);
+}
+```
+
+***NOTE
+`Alias::for()` will return `null` if no alias is found for the given class or object.
+***
```

Comment:
Do we really need this note? I think the code is pretty clear

Reply:
Hm, I think it still might be useful for newbies